logger/src/lib.rs

```diff
@@ -56,13 +57,19 @@ impl<D> Logger for Service<D>
 where
     D: Dal + Send + Sync + 'static,
 {
+    #[tracing::instrument(skip(self, request), fields(deployment_id = field::Empty, batch_size = field::Empty))]
     async fn store_logs(
         &self,
         request: Request<StoreLogsRequest>,
     ) -> Result<Response<StoreLogsResponse>, Status> {
         let request = request.into_inner();
         let logs = request.logs;
+
         if !logs.is_empty() {
+            Span::current().record("deployment_id", &logs[0].deployment_id);
+            Span::current().record("batch_size", logs.len());
```

Comment:
Maybe store the result of `Span::current()` so that we don't call it multiple times.